src/flask/cli.py

```diff
@@ -201,13 +201,10 @@ def prepare_import(path):
     module_name = []
 
     # move up until outside package structure (no __init__.py)
-    while True:
+    while os.path.exists(os.path.join(path, "__init__.py")):
```

Comment:
This change may impact functionnality, considering that lines 205 and 206 will not be executed prior to the condition being evaluated. The loop behaviours goes from a do...while loop to a while loop.